packages/core/database/src/index.ts

```diff
@@ -38,18 +39,19 @@ class Database {
 
   metadata: Metadata;
 
-  schema: SchemaProvider;
+  schema?: SchemaProvider;
 
-  migrations: MigrationProvider;
+  migrations?: MigrationProvider;
 
-  lifecycles: LifecycleProvider;
+  lifecycles?: LifecycleProvider;
 
-  entityManager: EntityManager;
+  entityManager?: EntityManager;
 
   static transformContentTypes = transformContentTypes;
 
   static async init(config: DatabaseConfig) {
     const db = new Database(config);
+    await db.setup();
     await validateDatabase(db);
     return db;
   }
```

Comment:
@innerdvations If we do not use an async constructor, then we need to setup the db asynchorizely, which cause the `entityManager` need to be decleared nullable, while the `entityManager` is used widely in the project and brings more things need to be updated.

Reply:
The root cause is that `is-using-typescript` checks `tsconfig.json` asynchronously: 

```js
module.exports = (dir, filename = undefined) => {
  const filePath = getConfigPath(dir, { filename });

  return fse.pathExists(filePath); // Promise<boolean>
};
```

Any idea about replace `pathExists` with `pathExistsSync` ?

Reply:
Making path synchronous is a workaround I'd rather avoid. If we can make it work "properly" with everything relying on an async action happening in/after the setup, that would be best. We can keep everything that works synchronously in the constructor, and then move everything after that to the async setup method and await it in init.

I will try to give this PR some closer attention this week :)